src/ToolBox/SOS/lldbplugin/CMakeLists.txt

```diff
@@ -124,7 +124,7 @@ endif()
 
 message(STATUS "LLDB_H: ${LLDB_H}")
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-delete-non-virtual-dtor")
+add_compile_options(-Wno-delete-non-virtual-dtor)
```

Comment:
This (and other similar places) essentially reverts a change that was made intentionally for compilation with GCC in #22687.

Reply:
What's the specific reasoning that we need to use `CMAKE_CXX_FLAGS` instead of `add_compile_options`? (for example we need to be able to remove these options in lower directories) I'm fine switching it back, but I want to make sure we have a reason why we're being inconsistent.

Reply:
I believe the reason was that these options apply only to C++ compiler and the gcc C compiler complains or fails on those. @franksinankaya am I correct that that was the reason why you've made this change?

Reply:
I just saw the reasoning (GCC for C specifically gets unhappy with some of the warning flags). I'll wait on this portion of the cleanup until we move to a version of CMake with the `$<COMPILE_LANGUAGE>` generator expression.

Reply:
Reasoning is correct.